src/bin/hal/cmd/bech32/encode.rs

```diff
@@ -0,0 +1,52 @@
+use bech32lib::{encode, CheckBase32, FromBase32, ToBase32};
+use clap;
+use hex;
+
+use cmd;
+use hal;
+
+pub fn subcommand<'a>() -> clap::App<'a, 'a> {
+	cmd::subcommand("encode", "encode bech32 format").args(&[
+		cmd::opt_yaml(),
+		cmd::opt("no-convert", "Do not convert payload to base32").required(false),
+		cmd::arg("hrp", "human-readable part").takes_value(true).required(true),
+		cmd::arg(
+			"payload-hex",
+			"hex-encoded payload bytes, 8-bit values\nunless --no-convert is specified",
+		)
+		.takes_value(true)
+		.required(true),
+	])
+}
+
+pub fn execute<'a>(matches: &clap::ArgMatches<'a>) {
+	let hrp = matches.value_of("hrp").expect("missing required argument");
+	let hex = matches.value_of("payload-hex").expect("missing required argument");
+
+	let payload: Vec<u8> = hex::decode(hex).expect("Invalid hex");
+
+	let payload_base32 = if matches.is_present("no-convert") {
+		payload.check_base32().expect("Invalid base32 payload")
```

Comment:
I prefer to have expect's with lowercase strings for consistency with the rest of the codebase.